### ISSUE TITLE
pyload won't start after installation

### ISSUE THREAD

**Issue by Hamm3rhart** [pyload/pyload#3428]
### Describe the problem

I can't start pyload after installation. I used this command to install it:
`pip3 install pyload-ng`
To start it i wrote:
`pyload`
but when i try to start it i get the followed output.
### Console Log

```
ModuleNotFoundError: No module named 'filetype'
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/pyload/core/utils/fs.py", line 18, in <module>
    import magic
ModuleNotFoundError: No module named 'magic'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/pyload", line 7, in <module>
    from pyload.script import main
  File "/usr/local/lib/python3.6/dist-packages/pyload/script.py", line 9, in <module>
    from .core import Core
  File "/usr/local/lib/python3.6/dist-packages/pyload/core/__init__.py", line 22, in <module>
    from .utils import format, fs
  File "/usr/local/lib/python3.6/dist-packages/pyload/core/utils/format.py", line 9, in <module>
    from .fs import fullpath
  File "/usr/local/lib/python3.6/dist-packages/pyload/core/utils/fs.py", line 21, in <module>
    from filetype import guess_mime
ModuleNotFoundError: No module named 'filetype'
```

### Additional references

System: Ubuntu 18.04.3 LTS

### Last Words

If you need some more information please tell me what. I will provide it to you. I hope you can help me.

**Comment by GammaC0de** [comment]
'pyload-ng' is not ready yet for production use.

You have to install `pyload` from the `stable` branch.
What OS are you using?

**Comment by Hamm3rhart** [comment]
@GammaC0de using Linux Ubuntu 18.04.3 LTS

**Comment by GammaC0de** [comment]
What you have to do is:
* install python 2.7, python-pycurl, pycrypto, pillow, tesseract
* install libmozjs-24-bin and make a symbolic link from /usr/bin/js24 to /usr/bin/js
* Download and extract https://github.com/pyload/pyload/archive/stable.zip to your installation folder.
* run pyLoadCore.py with python 2.7

**Comment by Hamm3rhart** [comment]
@GammaC0de can't install libmozjs-24-bin, using `apt-get install libmozjs-24-bin` i got this error.
```

Reading package lists... Done
Building dependency tree
Reading state information... Done
E: Unable to locate package libmozjs-24-bin
```

**Comment by GammaC0de** [comment]
Sorry, please install rhino instead.

**Comment by alpe12** [comment]
> 
> 
> @GammaC0de can't install libmozjs-24-bin, using `apt-get install libmozjs-24-bin` i got this error.
> 
> ```
> Building dependency tree
> Reading state information... Done
> E: Unable to locate package libmozjs-24-bin
> ```

I install libmozjs-52-dev on my sbc.

**Comment by GammaC0de** [comment]
Ok, so this issue can be closed, right?

**Comment by ErnestoD** [comment]
I am having similar issues but with Windows 10. I had Python 3 but installed Python 2 for this. When I run the pip it wants to install in my Python 3 directory. How do I get it to install into Python 2?

**Comment by GammaC0de** [comment]
@ErnestoD install pycurl and pycrypto via pip and then download an extract the latest stable from https://github.com/pyload/pyload/archive/stable.zip
After that run pyload via pyLoadCore.py

**Comment by dausruddin** [comment]
Sorry @GammaC0de , But I am so dead confused. Which one is the correct way to install?

Front page is showing installation for development version.

https://pyload.net/ is showing more things and said missing package can be installed from pip which some of them doesn't exist such as SafeEval, rename_process, PIL, node.js.

Then you gave us a simpler installation method here.

Which one should I use, and why they are not made clear enough in the readme?

Thanks

**Comment by GammaC0de** [comment]
@ErnestoD Use pip2 command to make sure you get it to install into Python 2.


@dausruddin What you have to do is (all other instructions are irrelevant):
* install python 2.7, python-pycurl, pycrypto, pillow, tesseract, nodejs
* Download and extract https://github.com/pyload/pyload/archive/stable.zip to your installation folder.
* run pyLoadCore.py with python 2.7

**Comment by MrQuestionnaire** [comment]
I'm installed ubuntu 18.04 (Armbian_19.11.3_Bananapi_bionic_current_5.3.9) on my bananapi M1.
After basic configuration I tried to set up pyload, as it was before on 16.04,04 TLS.
A lot of former necessay libs are not longer supported (qt4, python-imaging).

Lucky, I found this thread here. So I set up pyload with
cd /opt
git clone https://github.com/pyload/pyload.git
cd /pyload
git checkout stable

Before I try "pyLoadCore -s" I thought it would be a great idea, to install the new dependables as listed by GammaC0de.
None of the given commands succeeded. But it went better after installed setuptools and wheel.
pip2 install wheel
pip2 install setuptools

Now, at least installation of pycrypto, tesseract, nodejs and ffmpeg succeeded without errors.

pip2 install pillow

succeeded after installed missing libjpeg-dev
apt-get -y install libjpeg-dev
 
But for:  

pip2 install python-pycurl

I get this output:

Collecting python-pycurl
Exception:
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/usr/lib/python2.7/dist-packages/pip/commands/install.py", line 353, in run
    wb.build(autobuilding=True)
  File "/usr/lib/python2.7/dist-packages/pip/wheel.py", line 749, in build
    self.requirement_set.prepare_files(self.finder)
  File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 380, in prepare_files
    ignore_dependencies=self.ignore_dependencies))
  File "/usr/lib/python2.7/dist-packages/pip/req/req_set.py", line 554, in _prepare_file
    require_hashes
  File "/usr/lib/python2.7/dist-packages/pip/req/req_install.py", line 278, in populate_link
    self.link = finder.find_requirement(self, upgrade)
  File "/usr/lib/python2.7/dist-packages/pip/index.py", line 465, in find_requirement
    all_candidates = self.find_all_candidates(req.name)
  File "/usr/lib/python2.7/dist-packages/pip/index.py", line 423, in find_all_candidates
    for page in self._get_pages(url_locations, project_name):
  File "/usr/lib/python2.7/dist-packages/pip/index.py", line 568, in _get_pages
    page = self._get_page(location)
  File "/usr/lib/python2.7/dist-packages/pip/index.py", line 683, in _get_page
    return HTMLPage.get_page(link, session=self.session)
  File "/usr/lib/python2.7/dist-packages/pip/index.py", line 795, in get_page
    resp.raise_for_status()
  File "/usr/share/python-wheels/requests-2.18.4-py2.py3-none-any.whl/requests/models.py", line 935, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
HTTPError: 404 Client Error: Not Found for url: https://pypi.org/simple/python-pycurl/

I guess, without having python-pycurl installed, pyload will not work. I have no idea where to find the missing file?, bug?, configuration file? or whatever else causes the output above.

Is someone able and willing to help me? Please.

Thanks you for reading.

**Comment by GammaC0de** [comment]
@MrQuestionnaire 
`sudo apt install python-pycurl`

**Comment by MrQuestionnaire** [comment]
@GammaC0de: Thank you very much.

After sending last comment, I did a "pyLoadCore -s" regardless the missing "python-pycurl", but everything was working as it should be. 

After reading GammaC0des solve above, and doing "apt install python-pycurl" on console, I saw the output "python-pycurl is already the newest version". Now I understood that everything was place, and therefore it's not a miracle that my pi runs pyload for more than 1 day.  

Yep, looks like I am an idiot, trying to a install a OS - package via pip!

One thing for sure: the documentation (at least for installation) has to be reviewed and updated.
